tests/db/test_bigchain_api.py

```diff
@@ -7,196 +7,68 @@
                                             asset_id=tx.id)
         transfer_tx2 = transfer_tx2.sign([alice.private_key])
-        block3 = b.create_block([transfer_tx2])
-        b.write_block(block3)
 
-        # Vote both block2 and block3 valid
-        vote = b.vote(block2.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-        vote = b.vote(block3.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
+        with pytest.raises(DoubleSpend):
+            b.validate_transaction(transfer_tx2, [transfer_tx])
 
-        with pytest.raises(CriticalDoubleSpend):
-            b.get_spent(tx.id, 0)
+        b.store_bulk_transactions([transfer_tx])
 
-    @pytest.mark.genesis
-    def test_get_block_status_for_tx_with_double_inclusion(self, b, monkeypatch, alice):
-        from bigchaindb.exceptions import CriticalDoubleInclusion
+        with pytest.raises(DoubleSpend):
+            b.validate_transaction(transfer_tx2)
+
+    @pytest.mark.tendermint
+    def test_get_block_status_for_tx_with_double_inclusion(self, b, alice):
         from bigchaindb.models import Transaction
+        from bigchaindb.backend.exceptions import OperationError
 
         tx = Transaction.create([alice.public_key], [([alice.public_key], 1)])
         tx = tx.sign([alice.private_key])
 
-        monkeypatch.setattr('time.time', lambda: 1000000000)
-        block1 = b.create_block([tx])
-        b.write_block(block1)
-
-        monkeypatch.setattr('time.time', lambda: 1000000020)
-        block2 = b.create_block([tx])
-        b.write_block(block2)
-
-        # Vote both blocks valid (creating a double spend)
-        vote = b.vote(block1.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-        vote = b.vote(block2.id, b.get_last_voted_block().id, True)
-        b.write_vote(vote)
-
-        with pytest.raises(CriticalDoubleInclusion):
-            b.get_blocks_status_containing_tx(tx.id)
-
-    @pytest.mark.genesis
-    def test_get_transaction_in_invalid_and_valid_block(self, monkeypatch, b, alice):
-        from bigchaindb.models import Transaction
-
-        monkeypatch.setattr('time.time', lambda: 1000000000)
-        tx1 = Transaction.create([alice.public_key], [([alice.public_key], 1)],
-                                 metadata={'msg': 1})
-        tx1 = tx1.sign([alice.private_key])
-        block1 = b.create_block([tx1])
-        b.write_block(block1)
-
-        monkeypatch.setattr('time.time', lambda: 1000000020)
-        tx2 = Transaction.create([alice.public_key], [([alice.public_key], 1)],
-                                 metadata={'msg': 2})
-        tx2 = tx2.sign([alice.private_key])
-        block2 = b.create_block([tx2])
-        b.write_block(block2)
+        b.store_bulk_transactions([tx])
 
-        # vote the first block invalid
-        vote = b.vote(block1.id, b.get_last_voted_block().id, False)
-        b.write_vote(vote)
+        with pytest.raises(OperationError):
```

Comment:
Does this test differ somehow from the one before the one above?

Reply:
Other than that the first test is for a transfer and this is for a new asset, no. I don't know if that's a significant enough difference to justify a second test, but it's already written, so...

Reply:
It seems that this test tries to brute force storing a duplicate transaction and the above one tries to check if the double spend transaction in fact raises an exception when `validate_transaction` is called.